var/spack/repos/builtin/packages/flang/package.py

```diff
@@ -20,13 +20,9 @@ class Flang(CMakePackage):
     version('20180921', '4440ed5fdc390e4b7a085fb77b44ac54')
     version('20180612', '62284e26214eaaff261a922c67f6878c')
 
-    depends_on('llvm@flang-develop', when='@develop')
-    depends_on('llvm@flang-20180921', when='@20180921 target=x86_64')
-    depends_on('llvm@flang-20180612', when='@20180612 target=x86_64')
-
-    # LLVM version specific to OpenPOWER.
-    depends_on('llvm@flang-ppc64le-20180921', when='@20180921 target=ppc64le')
-    depends_on('llvm@flang-ppc64le-20180612', when='@20180612 target=ppc64le')
+    depends_on('llvm@develop+flang~gold~lldb~lld~compiler-rt~libcxx~polly~internal_unwind', when='@develop')
+    depends_on('llvm@7.0.1+flang~gold~lldb~lld~compiler-rt~libcxx~polly~internal_unwind', when='@20180921')
+    depends_on('llvm@6.0.0+flang~gold~lldb~lld~compiler-rt~libcxx~polly~internal_unwind', when='@20180612')
```

Comment:
My only problem with this syntax is that if someone already has `llvm+flang+anything_else`  installed, Spack will be forced to reinstall `llvm+flang~anything_else`. The concretizer still sucks at reusing installed software, so I'm fine with keeping this as is, but in the future we may want to change this to `llvm+flang`.

Reply:
I am not sure I understand. Do you mean?
`spack spec llvm@develop+flang`
vs.
`spack spec llvm+flang@develop`
are not the same thing to the concretizer?

I normally use the later syntax because it's less of an eyesore, but aren't these supposed to be equivalent?

Reply:
What I'm saying is that if I have `llvm@develop+flang+gold` or `llvm@develop+flang+libcxx` installed, this package will force Spack to reinstall `llvm@develop+flang~gold~libcxx`, even though either of those installations should work fine as dependencies.

Reply:
Right.  I was trying to be explicit, but it does prevent using a usable installed llvm+flang.  I will make the change and rebase this pull request.